File: js/src/maplestory/api/dto/character/characterAbilityInfoDto.ts
import { CharacterAbilityInfoDtoBody } from '@src/maplestory/api/response/character/characterAbilityDtoBody';

/**
 * 캐릭터 어빌리티 상세 정보
 */
class CharacterAbilityInfoDto {
  /**
   * 어빌리티 번호
   */
  abilityNo: string;

  /**
   * 어빌리티 등급
   */
  abilityGrade: string;

  /**
   * 어빌리티 옵션 및 수치
   */
  abilityValue: string;

  constructor(obj: CharacterAbilityInfoDtoBody) {
    const { ability_no, ability_grade, ability_value } = obj;

    this.abilityNo = ability_no;
    this.abilityGrade = ability_grade;
    this.abilityValue = ability_value;
  }
}

export { CharacterAbilityInfoDto };

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterAbilityInfoDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * 能力資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterAbilityInfoDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterAbilityInfoDTO {

    /**
     * 能力編號
     */
    @JsonProperty("ability_no")
    private String abilityNo;

    /**
     * 能力階級
     */
    @JsonProperty("ability_grade")
    private String abilityGrade;

    /**
     * 能力選項與數值
     */
    @JsonProperty("ability_value")
    private String abilityValue;
}
